### ISSUE TITLE
Tests failing for cuda-gen backend.

### ISSUE THREAD

**Issue by YohannDudouit** [CEED/libCEED#621]
t502, t506, t507, t537, t550, t552 are currently failing for the `cuda-gen` backend on the `main` branch.

**Comment by YohannDudouit** [comment]
```
0f54b25e60fd522f688efb036ba1cac9e3fcde63 is the first bad commit
commit 0f54b25e60fd522f688efb036ba1cac9e3fcde63
Author: jeremylt <thompson.jeremy.luke@gmail.com>
Date:   Tue Aug 25 10:28:49 2020 -0600

    CUDA - clean up logic for collograd in cuda/gen, should be based on all bases with interp/grad
```
My bad I approved this.

**Comment by jeremylt** [comment]
Huh, I had tested it locally on my machine

**Comment by jeremylt** [comment]
Ah, its dropping the double negation
75c7b2080cf69f285fbdb78ddf9afe348969eb95
reverting that alone will fix the problem.

**Comment by jeremylt** [comment]
nope, not quite

**Comment by jeremylt** [comment]
If I go into your branch, use f7eb965ac7e108a7c57610ce6b3f3fccec58b347, and comment out lines 815 and 856 of `ceed-cuda-gen-operator-build.cpp`, then everything works.

**Comment by YohannDudouit** [comment]
Lines 815 and 856?
`// Collect dim and Q1d`
`code << "\n#define CEED_Q_VLA "<<Q1d<<"\n\n";`

**Comment by jeremylt** [comment]
Lines 815 and 856 on that commit:
```
      if (emode == CEED_EVAL_GRAD)
```

**Comment by YohannDudouit** [comment]
Maybe that we want to fix this before #620 gets merged, it's already affecting people on the LLNL side.
